### PR TITLE
chore: add cdw as destinations

### PR DESCRIPTION
**Pull request by tomkit**
- add cdw destinations for out-of-band syncs

Screenshot 2023-11-17 at 2 28 23 PM
Screenshot 2023-11-17 at 2 28 31 PM

## Test Plan

- save and re-save redshift, bigquery, snowflake

## Deployment instructions

[Add any special deployment instructions here]

### CODE REVIEW

File: packages/core/mappers/destination.ts [hunk 1]
```diff
@@ -13,6 +21,27 @@ export const fromDestinationModelToUnsafe = async (model: DestinationModel): Pro
   const decryptedConfig = JSON.parse(await decrypt(model.encryptedConfig));
 
   switch (model.type) {
+    case 'bigquery':
+      return {
+        ...baseParams,
+        type: 'bigquery',
+        config: {
+          ...decryptedConfig,
+          credentials: camelcaseKeys(decryptedConfig.credentials),
```
Comment (asdfryan) [inline packages/core/mappers/destination.ts:30]: do we not need to do this for the other 2?

File: packages/core/services/destination_service.ts [hunk 2]
```diff
@@ -273,6 +278,33 @@ function mergeDestinationConfig(
         ...params.config,
         password: params.config.password ?? existingDestination.config.password,
       };
+    case 'bigquery':
```
Comment (asdfryan) [inline packages/core/services/destination_service.ts:281]: Can we just check `if (params.type !== existingDestination.type)` at the beginning

File: apps/mgmt-ui/src/components/connectors/destination/SnowflakeDestinationDetailsPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+    setDatabase(destination.config.database);
+    setSchema(destination.config.schema);
+    setUsername(destination.config.username);
+  }, [destination?.id]);
+
+  const createOrUpdateDestination = async (): Promise<DestinationSafeAny | undefined> => {
+    if (destination) {
+      const response = await updateDestination({
+        ...destination,
+        type: 'snowflake',
```
Comment (asdfryan) [inline apps/mgmt-ui/src/components/connectors/destination/SnowflakeDestinationDetailsPanel.tsx:67]: This looks incorrect

File: apps/mgmt-ui/src/components/connectors/destination/RedshiftDestinationDetailsPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+        </Stack>
+
+        <Stack className="gap-2">
+          <Typography variant="subtitle1">S3 Access Key</Typography>
+          <TextField
+            required={true}
+            error={!isNew && s3AccessKey === ''}
+            value={s3AccessKey}
+            size="small"
+            label="S3 Access Key"
```
Comment (asdfryan) [inline apps/mgmt-ui/src/components/connectors/destination/RedshiftDestinationDetailsPanel.tsx:376]: we should `type="password"` for this IMO

File: apps/mgmt-ui/src/components/connectors/destination/SnowflakeDestinationDetailsPanel.tsx [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+        <Stack className="gap-2">
+          <Typography variant="subtitle1">Password</Typography>
+          <TextField
+            required={true}
+            error={!isNew && password === ''}
+            value={password}
+            size="small"
+            label="Password"
+            variant="outlined"
+            helperText=""
```
Comment (asdfryan) [inline apps/mgmt-ui/src/components/connectors/destination/SnowflakeDestinationDetailsPanel.tsx:327]: we should `type="password"` for this IMO

File: apps/mgmt-ui/src/components/syncs/syncConfig/SyncConfigDetailsPanel.tsx [hunk 6]
```diff
@@ -185,7 +185,15 @@ function SyncConfigDetailsPanelImpl({ syncConfigId }: SyncConfigDetailsPanelImpl
...
 
-  const commonObjectsSupported = selectedProvider?.category !== 'no_category';
+  const selectedDestination = destinations.find((destination: any) => {
+    const isManagedDestination = destination.type === 'supaglue' && destinationName === 'Supaglue Managed Destination'; // this logic is a bit hairy because "supaglue" destination type has no "name" field
+    const isMatchedDestination = destination.name === destinationName;
+    return isMatchedDestination || isManagedDestination;
+  });
+
+  // Common Objects are only supported for categorized providers and Postgres destination
+  const isPostgresDestination = selectedDestination?.type === 'postgres' || selectedDestination?.type === 'supaglue';
```
Comment (asdfryan) [inline apps/mgmt-ui/src/components/syncs/syncConfig/SyncConfigDetailsPanel.tsx:195]: This is confusingly named

File: packages/core/services/destination_service.ts (hunk 2 above)
Comment (tomkit) [reply packages/core/services/destination_service.ts:281]: Just tried- it looks like the type narrowing doesn't work

File: packages/core/mappers/destination.ts (hunk 1 above)
Comment (tomkit) [reply packages/core/mappers/destination.ts:30]: I destructure and remove the sensitive password

**Review by asdfryan — APPROVED** [review]

**Pull request merged**